cupyx/scipy/stats/_stats_py.py

```diff
@@ -0,0 +1,143 @@
+import cupy
+
+
+def _contains_nan(a, nan_policy='propagate'):
+    policies = ['propagate', 'raise', 'omit']
+
+    if nan_policy not in policies:
+        raise ValueError("nan_policy must be one of {%s}" %
+                         ', '.join("'%s'" % s for s in policies))
+
+    contains_nan = cupy.isnan(cupy.sum(a))
+
+    if nan_policy == 'raise' and contains_nan:
+        raise ValueError("The input contains nan values")
```

Comment:
Let's compute contains_nan only when nan_policy is `raise`.

Reply:
Sure, I've updated it. I think we now don't need to define that in another function. Have changed it. Hope it's okay!